tests/e2e/specs/widgetPanel.js

```diff
@@ -154,4 +172,88 @@ describe("Widget panel usability test", () => {
     cy.get(`[data-cy=widget-delete-btn-yes]`).first().click({ force: true });
     cy.get(`[data-cy=widget-Misnamed]`).should("not.exist");
   });
+  it("Add Co2 widget & don't delete", () => {
+    cy.visit("/");
+    cy.get(`[data-cy=add-widget-btn]`).click();
+    cy.get(`[data-cy=widgetPanel-Heaviest-btn]`).contains("Top Heaviest Files");
+    cy.get(`[data-cy=widgetPanel-Unused-btn]`).contains("Top Unused Files");
+    cy.get(`[data-cy=widgetPanel-Misnamed-btn]`).contains("Top Misnamed Files");
+    cy.get(`[data-cy=widgetPanel-Co2-btn]`).contains("Carbon Emissions");
+    cy.get(`[data-cy=widgetPanel-Directory-btn]`).contains("Folder tracker");
+    cy.get(`[data-cy=widgetPanel-Co2-btn]`).click();
+    cy.get(`[data-cy=widget-small-size-btn]`).contains("3x3");
+    cy.get(`[data-cy=widget-large-size-btn]`).contains("6x6");
+    cy.get(`[data-cy=widget-small-size-btn]`).click();
+    cy.get(`[data-cy=widget-Co2]`).contains("Carbon Emissions");
+
+    longPress(`[data-cy=widget-Co2]`);
+    cy.get(`[data-cy=widget-delete-btn-yes]`).contains("yes");
+    cy.get(`[data-cy=widget-delete-btn-no]`).contains("no");
+    cy.get(`[data-cy=widget-delete-btn-no]`).first().click({ force: true });
+
+    cy.get(`[data-cy=widget-Co2]`).contains("Carbon Emissions");
+  });
+
+  it("Add Co2 widget & delete", () => {
+    cy.visit("/");
+    cy.get(`[data-cy=add-widget-btn]`).click();
+    cy.get(`[data-cy=widgetPanel-Heaviest-btn]`).contains("Top Heaviest Files");
+    cy.get(`[data-cy=widgetPanel-Unused-btn]`).contains("Top Unused Files");
+    cy.get(`[data-cy=widgetPanel-Misnamed-btn]`).contains("Top Misnamed Files");
+    cy.get(`[data-cy=widgetPanel-Co2-btn]`).contains("Carbon Emissions");
+    cy.get(`[data-cy=widgetPanel-Directory-btn]`).contains("Folder tracker");
+    cy.get(`[data-cy=widgetPanel-Co2-btn]`).click();
+    cy.get(`[data-cy=widget-small-size-btn]`).contains("3x3");
+    cy.get(`[data-cy=widget-large-size-btn]`).contains("6x6");
+    cy.get(`[data-cy=widget-small-size-btn]`).click();
+    cy.get(`[data-cy=widget-Co2]`).contains("Carbon Emissions");
+
+    longPress(`[data-cy=widget-Co2]`);
+    cy.get(`[data-cy=widget-delete-btn-yes]`).contains("yes");
+    cy.get(`[data-cy=widget-delete-btn-no]`).contains("no");
+    cy.get(`[data-cy=widget-delete-btn-yes]`).first().click({ force: true });
+    cy.get(`[data-cy=widget-Co2]`).should("not.exist");
+  });
+  it("Add Directory widget & don't delete", () => {
```

Comment:
you create a widget that lets you change folders, but you don't test it. Test the widget in its entirety.